In [1]:
import pandas as pd

In [2]:
caminho = r'..\data\ny_flights.csv'

In [3]:
df = pd.read_csv(caminho)

In [5]:
df.dropna(inplace=True)

In [10]:
subset = df.sample(n=1000)  # Randomly sample 1000 rows

In [12]:
from sklearn.gaussian_process import GaussianProcessRegressor
from sklearn.gaussian_process.kernels import RBF
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score
import numpy as np

# Assuming your DataFrame is named 'df' containing the flight data

# Extract features and target variable
# X = df[['CRS_DEP_TIME', 'DEP_TIME']]  # Add other relevant columns
X = subset[['CRS_DEP_TIME', 'DEP_TIME', 'TAXI_OUT', 'WHEELS_OFF', 'WHEELS_ON', 'TAXI_IN', 'CRS_ARR_TIME', 'CANCELLED', 'DIVERTED', 'CRS_ELAPSED_TIME', 'ACTUAL_ELAPSED_TIME', 'AIR_TIME', 'DISTANCE', 'CARRIER_DELAY', 'WEATHER_DELAY', 'NAS_DELAY', 'SECURITY_DELAY', 'LATE_AIRCRAFT_DELAY']]
y = subset['DEP_DELAY']

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Define the Gaussian Process regression model
kernel = RBF()
gpr = GaussianProcessRegressor(kernel=kernel)

# Train the model
gpr.fit(X_train, y_train)

# Make predictions on the testing set
y_pred = gpr.predict(X_test)

# Evaluate the performance of the model
mse = mean_squared_error(y_test, y_pred)
mae = mean_absolute_error(y_test, y_pred)
r2 = r2_score(y_test, y_pred)

print("Mean Squared Error:", mse)
print("Mean Absolute Error:", mae)
print("R-squared Score:", r2)


Mean Squared Error: 8577.82
Mean Absolute Error: 63.45
R-squared Score: -0.8239979416331802


1. Mean Squared Error (Erro Médio Quadrático): O valor obtido para o MSE é de 8577.82. Essa métrica mede a média dos erros ao quadrado entre os valores previstos e os valores reais do atraso de partida dos voos. Quanto menor o valor do MSE, melhor é o desempenho do modelo em fazer previsões precisas. No caso, um MSE de 8577.82 indica que as previsões têm uma quantidade razoável de erros ao quadrado em relação aos valores reais.

2. Mean Absolute Error (Erro Médio Absoluto): O valor obtido para o MAE é de 63.45. Essa métrica mede a média dos erros absolutos entre os valores previstos e os valores reais do atraso de partida dos voos. O MAE fornece uma medida da magnitude média dos erros de previsão, sem considerar a direção dos erros. Quanto menor o valor do MAE, melhor é o desempenho do modelo em fazer previsões precisas. No caso, um MAE de 63.45 indica que as previsões têm uma quantidade razoável de erros absolutos em relação aos valores reais.

3. R-squared Score (Coeficiente de Determinação): O valor obtido para o R-quadrado é de -0.8239979416331802. Essa métrica fornece uma medida da qualidade geral do ajuste do modelo aos dados. O R-quadrado varia de -∞ a 1, e quanto mais próximo de 1, melhor é o ajuste do modelo. No caso, um R-quadrado de -0.8239979416331802 indica que o modelo tem um ajuste ruim aos dados, pois o valor está negativo. Isso sugere que o modelo não é capaz de capturar as variações e tendências presentes nos dados.

Em resumo, os resultados indicam que o modelo de regressão Gaussian Process não está fornecendo previsões precisas e um bom ajuste aos dados do atraso de partida dos voos. É importante revisar e explorar outras abordagens de modelagem ou considerar a inclusão de mais recursos relevantes para melhorar o desempenho do modelo."